src/mlpack/tests/hyperplane_test.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file hyperplane_test.cpp
+ *
+ * Tests for Hyperplane and ProjVector implementations.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/tree/space_split/hyperplane.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::math;
+using namespace mlpack::tree;
+using namespace mlpack::metric;
+using namespace mlpack::bound;
+
+BOOST_AUTO_TEST_SUITE(HyperplaneTest);
+
+/**
+ * Ensure that a hyperplane, by default, consider all points to the left.
+ */
+BOOST_AUTO_TEST_CASE(HyperplaneEmptyConstructor)
+{
+  Hyperplane<EuclideanDistance> h1;
+  AxisOrthogonalHyperplane<EuclideanDistance> h2;
+
+  arma::mat dataset;
+  dataset.randu(3, 20); // 20 points in 3 dimensions.
+
+  for (size_t i = 0; i < dataset.n_cols; i++)
+  {
+    BOOST_REQUIRE(h1.Left(dataset.col(i)));
+    BOOST_REQUIRE(h2.Left(dataset.col(i)));
+    BOOST_REQUIRE(!h1.Right(dataset.col(i)));
+    BOOST_REQUIRE(!h2.Right(dataset.col(i)));
+  }
+}
+
+/**
+ * Ensure that we get the correct hyperplane given the projection vector.
+ */
+BOOST_AUTO_TEST_CASE(HyperplaneProjectionTest)
+{
+  // General hyperplane.
+  ProjVector projVect1(arma::vec("1 1"));
+  Hyperplane<EuclideanDistance> h1(projVect1, 0);
+
+  BOOST_REQUIRE_EQUAL(h1.Project(arma::vec("1 -1")), 0);
+  BOOST_REQUIRE(h1.Left(arma::vec("1 -1")));
+  BOOST_REQUIRE(!h1.Right(arma::vec("1 -1")));
+
+  BOOST_REQUIRE_EQUAL(h1.Project(arma::vec("-1 1")), 0);
+  BOOST_REQUIRE(h1.Left(arma::vec("-1 1")));
+  BOOST_REQUIRE(!h1.Right(arma::vec("-1 1")));
+
+  BOOST_REQUIRE_EQUAL(h1.Project(arma::vec("1 0")),
+      h1.Project(arma::vec("0 1")));
+  BOOST_REQUIRE(h1.Right(arma::vec("1 0")));
+  BOOST_REQUIRE(!h1.Left(arma::vec("1 0")));
+
+  BOOST_REQUIRE_EQUAL(h1.Project(arma::vec("-1 -1")),
+      h1.Project(arma::vec("-2 0")));
+  BOOST_REQUIRE(h1.Left(arma::vec("-1 -1")));
+  BOOST_REQUIRE(!h1.Right(arma::vec("-1 -1")));
+
+  // A simple 2-dimensional bound.
+  BallBound<EuclideanDistance> b1(2);
+
+  b1.Center() = arma::vec("-1 -1");
+  b1.Radius() = 1.41;
+  BOOST_REQUIRE(h1.Left(b1));
+  BOOST_REQUIRE(!h1.Right(b1));
+
+  b1.Center() = arma::vec("1 1");
+  b1.Radius() = 1.41;
+  BOOST_REQUIRE(h1.Right(b1));
+  BOOST_REQUIRE(!h1.Left(b1));
+
+  b1.Center() = arma::vec("0 0");
+  b1.Radius() = 1.41;
+  BOOST_REQUIRE(!h1.Right(b1));
+  BOOST_REQUIRE(!h1.Left(b1));
+
+  // AxisParallel hyperplane.
+  AxisParallelProjVector projVect2(1);
+  AxisOrthogonalHyperplane<EuclideanDistance> h2(projVect2, 1);
```

Comment:
I agree. Done in: https://github.com/mlpack/mlpack/pull/747/commits/207c1d428e3ca1028144369b98d15e5c7cb9abd4